wasmbinding/message_plugin.go

```diff
@@ -295,7 +294,40 @@ func (m *CustomMessenger) submitTx(ctx sdk.Context, contractAddr sdk.AccAddress,
 }
 
 func (m *CustomMessenger) submitAdminProposal(ctx sdk.Context, contractAddr sdk.AccAddress, submitAdminProposal *bindings.SubmitAdminProposal) ([]sdk.Event, [][]byte, error) {
-	response, err := m.performSubmitAdminProposal(ctx, contractAddr, submitAdminProposal)
+	var data []byte
+	err := m.validateProposalQty(&submitAdminProposal.AdminProposal)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to validate proposal quantity")
+	}
+	// here we handle pre-sdk47 style of proposals: param change & upgrade/cancel
+	if m.isLegacyProposal(&submitAdminProposal.AdminProposal) {
+		resp, err := m.performSubmitAdminProposalLegacy(ctx, contractAddr, submitAdminProposal)
+		if err != nil {
+			ctx.Logger().Debug("performSubmitAdminProposalLegacy: failed to submitAdminProposal",
+				"from_address", contractAddr.String(),
+				"creator", contractAddr.String(),
+				"error", err,
+			)
+			return nil, nil, errors.Wrap(err, "failed to submit admin proposal legacy")
+		}
+		data, err = json.Marshal(resp)
+		if err != nil {
+			ctx.Logger().Error("json.Marshal: failed to marshal submitAdminProposalLegacy response to JSON",
+				"from_address", contractAddr.String(),
+				"creator", contractAddr.String(),
+				"error", err,
+			)
+			return nil, nil, errors.Wrap(err, "marshal json failed")
```

Comment:
we never return any events from this func. can we remove the first return value?

Reply:
no, bc there is such requirement  in `DispatchMsg` func